### PR TITLE
[github] Manage private secrets with GitHub Secrets to enhance security

### PR DESCRIPTION
**Pull request by rtsisyk**
1. Restore the original simple version of `configure.sh`
    Restores 08e37f4 "Refactor configure.sh"
    Reverts b87ee95b "Fixed configure.sh script and gh actions"
   
2. Use GitHub Secrets instead of a private git repository to enhance security standards and ensure credentials are encrypted and safely managed.

3. Document credentials used by GitHub Actions in docs/CREDENTIALS.md

4. Include `network_security_config.xml` directly into the repo as it has nothing sensitive.

5. Include Apple WWDR intermediate certificates directly into the repo as they are not sensitive and publicly available. https://developer.apple.com/help/account/reference/wwdr-intermediate-certificates

6. Add `private.h` in the repository since it does not differ from
    `private_defaults.h`.

### CODE REVIEW

File: configure.sh [hunk 1]
```diff
@@ -1,171 +1,21 @@
...
-  echo "You need to have Boost submodule present to run bootstrap.sh"
-  echo "Try 'git submodule update --init --recursive'"
-  exit 1
-fi
-if [ ! -d "$BASE_PATH/3party/boost/tools" ]; then
-  echo "Boost's submodule 'tools' is missing, it is required for bootstrap"
-  echo "Try 'git submodule update --init --recursive'"
-  exit 1
+if [ ! -f private.h ]; then
+  cp -p private_default.h private.h
```
Comment (pastk) [inline configure.sh:11]: is it needed at all?

e.g. we don't keep `gradle.properties` in .gitignore, though some people might change it to suit their needs

File: configure.sh [hunk 2]
```diff
@@ -1,171 +1,21 @@
...
-  echo "For Windows please run:"
-  echo "cd 3party\\boost"
-  echo "bootstrap.bat"
-  echo "b2 headers"
-  echo "cd ..\\.."
-else
-  ./bootstrap.sh
-  ./b2 headers
+if [ ! -d 3party/boost/tools ]; then
+  git submodule update --init --recursive
```
Comment (pastk) [inline configure.sh:15]: nice!
Why we didn't add it earlier? :)

File: .github/workflows/android-beta.yaml [hunk 3]
```diff
@@ -56,19 +56,25 @@ jobs:
...
-
-      - name: Configure repo with private keys
+      - name: Restore beta keys
         shell: bash
         run: |
-          ./configure.sh ./private.git
-          rm -rf ./private.git
+          echo "$PRIVATE_H" | base64 -d > private.h
+          echo "$FIREBASE_APP_DISTRIBUTION_JSON" | base64 -d > android/app/firebase-app-distribution.json
+          echo "$GOOGLE_SERVICES_JSON" | base64 -d >android/app/google-services.json
```
Comment (pastk) [inline .github/workflows/android-beta.yaml:64]:
```suggestion
          echo "$GOOGLE_SERVICES_JSON" | base64 -d > android/app/google-services.json
```

**Review by pastk — APPROVED** [review]
LGTM

It'd be good to have a review from someone who knows what all these keys are for :)

**Comment by map-per** [comment]
Was there a specific reason for changing form `FdroidBeta` to `FdroidDebug`? I found it quite convenient that the CI used to build a beta version as they, unlike debug versions, use the real OSM database.

**Pull request merged**